libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp

```diff
@@ -1,267 +1,516 @@
+        free(newAP.ssid);
+        return false;
+    }
+
+    _APlist.push_back(newAP);
+    DEBUG_WIFI_MULTI("[WIFIM][APlistAdd] Add SSID: %s\n", newAP.ssid);
+    return true;
+}
+
+/**
+ * @brief Check if AP exists in list
+ * @param ssid
+ *      WiFi SSID
+ * @param passphrase
+ *      WiFi Password
+ * @return
+ */
+bool ESP8266WiFiMulti::APlistExists(const char *ssid, const char *passphrase)
+{
+    if (!ssid || (*ssid == 0x00) || (strlen(ssid) > 32)) {
+        // Fail SSID too long or missing
+        DEBUG_WIFI_MULTI("[WIFIM][APlistExists] No ssid or ssid too long\n");
+        return false;
+    }
+
+    for (auto entry : _APlist) {
+        if (!strcmp(entry.ssid, ssid)) {
+            if (!passphrase) {
+                if (!strcmp(entry.passphrase, "")) {
+                    return true;
+                }
+            } else {
+                if (!strcmp(entry.passphrase, passphrase)) {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+/**
+ * @brief Remove all AP's from list
+ */
+void ESP8266WiFiMulti::APlistClean(void)
+{
+    // Remove all entries from APlist
+    for (auto entry : _APlist) {
+        if (entry.ssid) {
+            free(entry.ssid);
+        }
+        if (entry.passphrase) {
+            free(entry.passphrase);
+        }
+    }
+
+    _APlist.clear();
+}
+
+/**
+ * @brief Print WiFi scan results
+ * @details
+ *      Macro DEBUG_ESP_WIFI and DEBUG_ESP_PORT must be configured
+ */
+void ESP8266WiFiMulti::printWiFiScan()
+{
+#ifdef DEBUG_ESP_WIFI
+    String ssid;
+    int32_t rssi;
+    uint8_t encryptionType;
+    uint8_t* bssid;
+    int32_t channel;
+    bool hidden;
+    int8_t scanResult;
+
+    scanResult = WiFi.scanComplete();
+
+    DEBUG_WIFI_MULTI("[WIFIM] %d networks found:\n", scanResult);
+
+    // Print unsorted scan results
+    for (int8_t i = 0; i < scanResult; i++) {
+        bool known = false;
+
+        WiFi.getNetworkInfo(i, ssid, encryptionType, rssi, bssid, channel, hidden);
+
+        for(auto entry : _APlist) {
+            if(ssid == entry.ssid) {
+                // SSID match
+                known = true;
+            }
+        }
+
+        if (known) {
+            DEBUG_WIFI_MULTI(" --->");
+        } else {
+            DEBUG_WIFI_MULTI("     ");
+        }
+
+        DEBUG_WIFI_MULTI(" %d: [CH %02d] [%02X:%02X:%02X:%02X:%02X:%02X] %ddBm %c %s\n",
+                         i,
+                         channel,
+                         bssid[0], bssid[1], bssid[2],
+                         bssid[3], bssid[4], bssid[5],
+                         rssi,
+                         (encryptionType == ENC_TYPE_NONE) ? ' ' : '*',
+                         ssid.c_str());
+        delay(0);
+    }
+#endif
+}
+
+/**
+ * @brief Print WiFi status
+ * @details
+ *      Macro DEBUG_ESP_WIFI and DEBUG_ESP_PORT must be configured
+ * @param status
+ *      WiFi status
+ */
+static void printWiFiStatus(wl_status_t status)
```

Comment:
Please move this to the top and remove the function declaration there. It's cleaner to have such utility private functions at the top.
That also mimics an anonymous namespace, so it's consistent.

Reply:
Done.